bin/opteadm/src/bin/opteadm.rs

```diff
@@ -283,6 +299,51 @@ impl TryFrom<SnatConfig> for SNat6Cfg {
     }
 }
 
+#[derive(Args, Debug)]
+struct DhcpConfig {
+    /// The hostname a connected guest should be provided via DHCP.
+    #[arg(long)]
+    hostname: Option<DomainName>,
+
+    /// The domain used by a guest to contruct its FQDN, provided via DHCP.
+    #[arg(long)]
+    host_domain: Option<DomainName>,
+
+    /// A comma-delimited list of DNS server IP addresses to provide over DHCP.
+    #[arg(long, value_delimiter = ',')]
+    dns_servers: Vec<IpAddr>,
+
+    /// A comma-separated list of domain names provided to the guest,
+    /// used when resolving hostnames.
+    #[arg(long, value_delimiter = ',')]
+    domain_search_list: Vec<DomainName>,
+}
+
+impl From<DhcpConfig> for DhcpCfg {
+    fn from(value: DhcpConfig) -> Self {
+        let mut dns4_servers = vec![];
+        let dns6_servers = value
+            .dns_servers
+            .into_iter()
+            .filter_map(|ip| match ip {
+                IpAddr::Ip4(ip) => {
+                    dns4_servers.push(ip);
+                    None
+                }
+                IpAddr::Ip6(ip) => Some(ip),
+            })
+            .collect();
```

Comment:
```suggestion
        let (dns4_servers, dns6_servers) = value
            .dns_servers
            .into_iter()
            .partition(|ip| matches!(ip, IpAddr::Ip4(_)))
            .collect();
```

Reply:
This is not quite the same; this results in `(Vec<IpAddr>, Vec<IpAddr>)`, but we need `(Vec<Ipv4Addr>, Vec<Ipv6Addr>)`. (I realized this while trying to implement the same thing in Omicron.)

Reply:
ah, d'oh. right